File: journeyrec-server/src/main/java/com/mottmac/journeyrec/domain/JourneyComposite.java
package com.mottmac.journeyrec.domain;

import java.util.List;

public class JourneyComposite {
    private Journey journey;
    private List<JourneyPoint> points;

    public JourneyComposite(Journey journey, List<JourneyPoint> points) {
        this.journey = journey;
        this.points = points;
    }

    public Journey getJourney() {
        return journey;
    }

    public void setJourney(Journey journey) {
        this.journey = journey;
    }

    public List<JourneyPoint> getPoints() {
        return points;
    }

    public void setPoints(List<JourneyPoint> points) {
        this.points = points;
    }
}

File: flutter_db_provider/lib/dao/journey_composite.dart

import 'dart:convert';

import 'journey.dart';
import 'journey_point.dart';

class JourneyComposite {

  Journey journey;
  List<JourneyPoint> journeyPoints;

  JourneyComposite(this.journey, this.journeyPoints);

  Map toJson() => {
    'journey': jsonEncode(journey),
    'points': jsonEncode(journeyPoints),
  };
}